### PR TITLE
Prepare yaru for incoming libadwaita changes

### LINKED ISSUE #2798: When wearing glasses with high correction, the closing button moves within the red circle
**Explanation/Summary**

When using the Yaru theme, the colour of the X within the close button differs from the background around the red circle. For people using glasses with high correction, this causes the X to move within the red circle based on the viewing angle. To resolve this issue, the colour of the X needs to be the same as the title bar background. Since this is the case when using yaru-light, that's a valid "workaround". If at all possible, without affecting your style and branding, I would like to be able to change the colour of the X within the red circle.

**Expected Behavior**

The closing button (X) remains in the centre of the closing button.

**Actual Behavior**

When wearing glasses, the closing button (X) moves within the red cricle.

**Steps to Reproduce the Problem**

This is going to seem silly:

1. Wear glasses
2. Use the default or dark yaru theme
3. look at the closing button and move your head from left to right

**Software that presents the issue**

All windows that use the yaru theme (and window borders)

**Yaru version**

* `20.04.10.1`
* `21.04.1`

Though, I'm fairly certain all version are affected.

**Screenshots**

Affected:

Screenshot from 2021-04-23 18-38-04

Not affected:

Screenshot from 2021-04-23 18-38-11

**Upstream check**

This issue does not affect Adwaita, as Adwaita doesn't use a circle around the closing button and the colour of the X and background differ.

### LINKED ISSUE #2925: Remove aubergine from the Yaru palette
This comes directly from a rebranding initiative at canonical:

All widgets coloured in aubergine need to be coloured in monochrome or ???. The only exception are the green elements.

This needs changes in gtk2, gtk3, gtk4, cursor theme and gnome shell theme for:

- sliders
- checks/radios
- switches
- the wait cursors

WIP: on waiting mode until the accent color is chosen

### PR DESCRIPTION
**Pull request by Feichtmeier**
- this removes aubergine from being used since libadwaita will only allow ubuntu to use one accent color
- libadwaita also has a gray close button, to not create inconsistency in the headerbar we use this in gtk3, too
- this also needs a change for the slider color to be some sort of white now, the warm_gray does not work inside an orange switch

TODO:

- [x] adapt gtk2
- [x] adapt icon theme (done in #3265)
- [ ] adapt unity, metacity, xfwm4 parts (_probably in another PR once this PR is finished_)

Closes #2925
Closes #2798

grafik
grafik
gtk4
grafik

### CODE REVIEW

**Comment by Jupi007** [comment]
Thanks @Feichtmeier 
Sad but needed changes. I will really miss our soft grey slide handle :cry:

We also need to adapt the icon theme, I can do it if you want.

File: gtk/src/default/gtk-4.0/_tweaks.scss [hunk 1]
```diff
@@ -23,16 +17,7 @@ windowcontrols {
...
-                    background: gtkmix($accent_bg_color, "#fff", 90%);
-                }
-
-                &:active {
-                    background: gtkmix($accent_bg_color, "#000", 90%);
-                }
+                background: $button_color;
 
                 &:backdrop {
                     filter: saturate(0);
```
Comment (Jupi007) [inline gtk/src/default/gtk-4.0/_tweaks.scss:23]: This line can be removed, it was needed for the orange close button to looks grey in backdrop.

File: gnome-shell/src/gnome-shell-sass/_palette.scss [hunk 2]
```diff
@@ -39,5 +39,5 @@ $darkblue: #335280;
 // Semantic colors
 $primary_accent_bg_color: $orange;
 $primary_accent_fg_color: $white;
-$secondary_accent_bg_color: $light_aubergine;
+$secondary_accent_bg_color: $orange;
```
Comment (Jupi007) [inline gnome-shell/src/gnome-shell-sass/_palette.scss:42]: This `secondary` accent system can also be removed imo.

**Comment by ItzSwirlz** [comment]
Will aubergine still be used outside of apps requiring/built on libadwaita?

**Comment by LinuxLlama** [comment]
I don't think gtk3 apps should switch to libadwaita style toggles. They don't fit in with the rest of the theme, especially with the shadow and lack of an outline. See example below:
Screenshot from 2021-12-13 17-49-42

**Comment by Feichtmeier** [comment]
> Will aubergine still be used outside of apps requiring/built on libadwaita?

In theory we could throw it out of the repo but on the other side it doesn't hurt anyone. I'll ask the others when I'm done with gtk2, I personally do not have an opinion

File: gnome-shell/src/gnome-shell-sass/_palette.scss (hunk 2 above)
Comment (Feichtmeier) [reply gnome-shell/src/gnome-shell-sass/_palette.scss:42]: Hmmm, two questions:

1) don't we need those colors somewhere for the yaru flavours?
2) if we remove the secondary accent color can't we remove that additional abstraction of accent colors alltogether and stick to selected bg color like upstream gtk3?

Comment (Jupi007) [reply gnome-shell/src/gnome-shell-sass/_palette.scss:42]: I would prefer keep this abstraction in the `_palette` file to keep the symlink system based flavour.
In my mind, I think we should rename both `$primary_accent_...` and `$secondary_accent_...` to just `$accent_...`.

**Comment by Feichtmeier** [comment]
@ubuntu/yaru-team 
should we also adapt the close button sizes to the a bit bigger size of gtk4?
It won't be as jarring as with the orange close button in that size I think

**Comment by Jupi007** [comment]
It's also my opinion.

Maybe also consider to totally remove `_tweaks` for Gtk4.

**Comment by Feichtmeier** [comment]
> It's also my opinion.
> 
> Maybe also consider to totally remove `_tweaks` for Gtk4.

the strange thing is that they add that soft gray to all windowcontrol button :D

**Comment by Jupi007** [comment]
I know... But it will looks like this with Libadwaita anyway so 😅

Maybe ask upstream ;)

**Comment by Muqtxdir** [comment]
> should we also adapt the close button sizes to the a bit bigger size of gtk4?
It won't be as jarring as with the orange close button in that size I think

+1 and also to maintain similarity to `libadwaita` maybe extend the background circle to minimize and maximize as well?

_**for reference screenshot:  GNOME text editor with libadwiata**_
libadwaita_app

also we have little less-rounded borders compared to `libadwaita` should that also be changed ?

**Comment by ItzSwirlz** [comment]
Other note: does this impact GtkSourceView and its files?

**Comment by Jupi007** [comment]
GtkSourceView is not impacted by the change. We still have control on that for now.

**Comment by Feichtmeier** [comment]
Full orange has been successfully deployed onto our eyes :eye: :eye: :orange_circle: 

grafik

**Review by Jupi007 — APPROVED** [review]
LGFM :+1:

Thanks @Feichtmeier

**Comment by Feichtmeier** [comment]
So metacity and the test still use purple?

**Comment by dominichayesferen** [comment]
My take: If you're have LibAdwaita references in your close titlebutton... either make it consistent with LibAdwaita's ALL TITLEBUTTONS have circles, or have none of them have circles.

I get you're trying to do a double reference of Yaru and LibAdwaita by making only close be the one with its circle always prominent, but honestly the way it is right now in this Pull Request, it's gonna make a lot of people think their close button is permanently hovered or pressed.

Honestly, overall I think it'd just be better if both this, and LibAdwaita upstream, dropped the prominent circles around titlebar buttons altogether at this point as it's just inconsistent overall with headerbar buttons appearance-behaviour wise, and like I said it might make people think their titlebuttons are permanently focused/pressed or hovered.

**Comment by Muqtxdir** [comment]
@dominichayesferen please see this [comment](https://github.com/ubuntu/yaru/pull/3267#issuecomment-996678294)

**Pull request merged**